lib/streams/function-stream.js

```diff
@@ -0,0 +1,55 @@
+var Readable = require('../readable');
+var inherits = require('util').inherits;
+
+/**
+ * FunctionStream takes a function with callbacks, which pushes BatchStream batches the incoming data into an array of size `batchSize`
+ * NOTE: objectMode will always be true
+ *
+ * @class FunctionStream
+ * @constructor
+ * @extends Transform
+ * @param {Function} func - Function to generate stream entries
+ * @param {Object} [options] - Stream options object
+ */
+function FunctionStream(func, options) {
+	options = !options ? {} : options;
+
+	options.objectMode = true;
+	Readable.call(this, options);
+
+	this._func = func;
+}
+inherits(FunctionStream, Readable);
+
+FunctionStream.prototype.callFunc = function(times) {
+	if(times <= 0) { return; }
+	var self = this;
+	this._func(function(error, entry) {
+		if(error) { return self.triggerChainError(error); }
```

Comment:
You should just self.emit('error') here, to allow for errors from this specific stream to be handled, instead of triggering chain errors immediately.  Chain errors are only triggered if an 'error' event is not handled.